protocols/v2/noise-sv2/src/signature_message.rs

```diff
@@ -24,6 +30,13 @@ impl From<[u8; 74]> for SignatureNoiseMessage {
 }
 
 impl SignatureNoiseMessage {
+    /// Verifies the `SignatureNoiseMessage` against the provided public key and an optional
+    /// authority public key. The verification checks that the message is currently valid
+    /// (i.e., withing the `valid_from` and `not_valid_after` time window) and that the signature
+    /// is correctly signed by the authority.
+    ///
+    /// If an authority public key is not provided, the function assumes that the signature
+    /// is already valid without further verification.
```

Comment:
```suggestion
    /// Verifies the Schnorr signature of the `SignatureNoiseMessage`.
    ///
    /// Checks whether the message's Schnorr signature is valid by verifying it against the
    /// provided public key (`pk`) and authority key (`authority_pk`). It also checks whether the
    /// message is within its validity period, ensuring that it is neither expired nor prematurely
    /// used.
    ///
    /// Returns `True` if the signature is valid and the message is within its validity period;
    /// otherwise, `False`.
```